routing/routing.go

```diff
@@ -154,19 +161,73 @@ func New(o Options) *Routing {
 
 	r := &Routing{log: o.Log, quit: make(chan struct{})}
 	initialMatcher, _ := newMatcher(nil, MatchingOptionsNone)
-	r.matcher.Store(initialMatcher)
+	rt := &routeTable{
+		m:       initialMatcher,
+		created: time.Now().UTC(),
+	}
+	r.routeTable.Store(rt)
 	r.startReceivingUpdates(o)
 	return r
 }
 
+// ServeHTTP renders the list of current routes.
+func (r *Routing) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	rt := r.routeTable.Load().(*routeTable)
+	req.ParseForm()
+	createdUnix := strconv.FormatInt(rt.created.Unix(), 10)
+	ts := req.Form.Get("timestamp")
+	if ts != "" {
+		if createdUnix != ts {
+			http.Error(w, "invalid timestamp", http.StatusBadRequest)
+			return
+		}
+	}
+	offset, err := extractParam(req, "offset", 0)
+	if err != nil {
+		http.Error(w, "invalid offset", http.StatusBadRequest)
+		return
+	}
+	limit, err := extractParam(req, "limit", defaultRouteListingLimit)
+	if err != nil {
+		http.Error(w, "invalid limit", http.StatusBadRequest)
+		return
+	}
+	accept := req.Header.Get("Accept")
+	routes := slice(rt.validRoutes, offset, limit)
+
+	// TODO: Maybe this is too strict?
+	if accept == "application/json" {
```

Comment:
i think the accept mime should work like this:

1. if it contains application/json, return json
2. if it contains application/eskip, plain/text, plain/*, */* or not set at all, return eskip
3. if it is set to anything else return 406